### PR TITLE
Replace Strings with Urls and Paths in deno_dir module resolution

### PR DESCRIPTION
**Pull request by NewLunarFire**

### CODE REVIEW

File: src/deno_dir.rs [hunk 1]
```diff
@@ -481,5 +481,24 @@ fn test_resolve_module() {
 const ASSET_PREFIX: &str = "/$asset$/";
 
 fn is_remote(module_name: &str) -> bool {
-  module_name.starts_with("http")
+  let url = Url::parse(module_name);
+    if url.is_err() {
```
Comment (ry) [inline src/deno_dir.rs]: There appears to be some bad indent here. Please run ./tools/format.py

File: src/deno_dir.rs [hunk 2]
```diff
@@ -481,5 +481,24 @@ fn test_resolve_module() {
 const ASSET_PREFIX: &str = "/$asset$/";
 
 fn is_remote(module_name: &str) -> bool {
-  module_name.starts_with("http")
+  let url = Url::parse(module_name);
```
Comment (ry) [inline src/deno_dir.rs]: Are we calling Url::parse twice now during CodeFetch ? :/

Comment (NewLunarFire) [reply src/deno_dir.rs]: We are calling it once on module_specifier in resolve_module, and once in this function, I think it's fine, this operation is nullipotent and both variables are string that can either be an Url or a local path.

We could change those to use Url instead, but Urls cannot be relative and atm, the code assumes they are strings. Maybe use a type that can be either an Url or a Path.

Or we can let the Url parsing here.

Comment (ry) [reply src/deno_dir.rs]: @NewLunarFire I'd prefer it if we could figure out how to not parse the URL twice.

**Comment by NewLunarFire** [comment]
@ry I reworked a lot more of the code. We don't parse the URL twice in deno_dir anymore. We still have to parse the url a second time in net because Hyper::Uri's object cannot be created from url::Url.

**Comment by ry** [comment]
@NewLunarFire Do you think it's possible to only use `Hyper::Uri` and remove `url::Url` that we're currently using? It seems unnecessary to have to much duplicated code.

**Comment by NewLunarFire** [comment]
@ry Based on what we use from url::Url, I think we could. Also, I wonder if this is the only location we use the url crate.

**Comment by ry** [comment]
I think so, but grep for it to make sure. If we can remove the url crate - I think that would be best.

**Comment by ry** [comment]
@NewLunarFire this one is also seriously out of date. I'm going to close it - but I'm still interested in the deno_dir refactor - specially if we can remove the url crate and only use hyper's uri. Please reopen a new PR if you rebase. Apologies and appreciation as always.

**Pull request closed without merging**